monitoring/uss_qualifier/scd/simulator/operational_intents_flight_authorisation_data_generator.py

```diff
@@ -0,0 +1,449 @@
+        
+        coin_flip = random.choice([0,0,1])         
+        if coin_flip:
+            flight_geometry = self._generate_random_flight_polygon()
+        else:
+            flight_geometry = self._generate_random_flight_path()
+
+        if injection_number == 0:
+            self.first_flight_geometry = asShape(flight_geometry)
+
+        if geometry_generation_rule.intersect_space: # This is not the first geometry, check if it should intersect with the control
+            geometry_intersects = False
+            while (geometry_intersects == False):
+                coin_flip = random.choice([0,0,1]) 
+                # We are trying to generate a path that intersects with the control, we keep generating linestrings or polygons till one is found that does intersect
+                if coin_flip:
+                    flight_geometry = self._generate_random_flight_polygon()
+                else:
+                    flight_geometry = self._generate_random_flight_path()
+                    
+                raw_geom = asShape(flight_geometry) # Generate a shape from the geometry
+                
+                geometry_intersects = self.first_flight_geometry.intersects(raw_geom) # Check this intersects with the control     
+            
+        return flight_geometry
+
+    def convert_geometry_to_volume(self, flight_geometry:LineString, altitude_of_ground_level_wgs_84:int) -> Volume3D:
+        ''' A method to convert a GeoJSON LineString or Polygon to a ASTM outline_polygon object by buffering 15m spatially '''
+        
+        flight_geometry_shp = asShape(flight_geometry)
+        flight_geometry_utm = self.utm_converter(flight_geometry_shp)
+        buffer_shape_utm = flight_geometry_utm.buffer(15)        
+
+        alt_upper = altitude_of_ground_level_wgs_84 + self.altitude_agl +self.altitude_envelope  
+        alt_lower = altitude_of_ground_level_wgs_84 + self.altitude_agl - self.altitude_envelope
+
+        buffered_shape_geo = self.utm_converter(buffer_shape_utm, inverse=True)
+        
+        all_vertices = []
+
+        altitude_upper = Altitude(value= alt_upper, reference = 'W84', units='M')
+        altitude_lower = Altitude(value=alt_lower, reference = 'W84', units='M')
+        for vertex in list(buffered_shape_geo.exterior.coords):
+            coord = LatLngPoint(lat = vertex[0] , lng = vertex[1])
+            all_vertices.append(coord)
+        
+        p = Polygon(vertices=all_vertices)
+        
+        volume3D = Volume3D(outline_polygon = p, altitude_lower = altitude_lower, altitude_upper = altitude_upper, outline_circle = {})
+        
+        return volume3D
+
+    def transform_3d_volume_to_astm_4d(self, volume_3d : Volume3D) -> Volume4D:
+        ''' This method converts a 3D Volume to 4D Volume, the flight start time is 3 mins from now  '''
+    
+        three_mins_from_now = self.now.shift(minutes = 3)
```

Comment:
When this FlightInjectionRequest is actually sent, this timestamp will be far, far in the past.  We probably need to add a field to FlightInjectionAttempt that will allow test executor to adjust the timestamps in the same way as we did with remote ID.

Reply:
Thanks! I have now added a `reference_time` and a `flight_plan_delay` to the definition and updated the simulator to add both.